src/platforms/javascript/common/session-replay/configuration.mdx

```diff
@@ -87,21 +88,57 @@ new Replay({
 
-Session Replay works by recording incremental DOM changes that occur in your web application. Generally these changes occur in small batches and cause minimal overhead. However, it can be easy to overlook cases that will cause a large amount of DOM mutations to happen in a single update. An example is a custom dropdown component that has an unbounded list of options to render. This can negatively impact performance without Session Replay and its effects will be magnified with Session Replay enabled. In order to avoid this scenario, Session Replay will stop recording if it detects a large number of mutations (default: 10,000), which can be configured by setting `mutationLimit`. Additionally, we provide breadcrumbs in the replay to warn you when a large number of mutations are detected (default: 750).
+You can use the Sentry SDK to link a user to a session. Read more about it in our <PlatformLink to="/enriching-events/identify-user">docs</PlatformLink>.
+
+```javascript
+Sentry.setUser({ email: "jane.doe@example.com" });
+```
+
+## Using a Custom Compression Worker
+
+By default, Session Replay will use an inlined web worker script to compress Replay data before we send it over the network. This helps to drastically reduce the amount of data transferred, improving performance and reducing network overhead. Because compressing data is CPU intensive, we use a web worker to offload this work to a separate thread.
+
+While an inlined worker will work well for most applications, there are two main problems with this:
+
+1. Since the worker code is inlined, it increases the main bundle size of your application.
+2. Using inline workers can cause CSP violations if your application has a very strict CSP policy. [Learn more about configuring your CSP policy.](../#content-security-policy-csp)
+
+To solve these problems, we provide the ability to use a custom worker script. This allows you to host the worker script on your own domain, and avoid the CSP issues that come with inline workers. Additionally, the worker script can then be removed from the main application bundle, and can be served and cached separately.
+
+Follow these steps in order to use a custom compression worker:
+
+1. Download the [example worker script](https://github.com/getsentry/sentry-javascript/blob/develop/packages/replay-worker/examples/worker.min.js) from our Github repository.
+2. Host this worker script on the same domain as your application. For example, if your application is hosted at `https://example.com`, you could host the worker script at `https://example.com/assets/worker.min.js`.
+3. Configure your custom worker script in replay:
 
 ```javascript
 new Replay({
-  mutationBreadcrumbLimit: 1000,
-  mutationLimit: 1500,
+  workerUrl: "/assets/worker.min.js",
 });
 ```
 
-## Identifying Users
+4. You can now [tree shake](../../configuration/tree-shaking/) the default included worker script. If you are using our bundler plugins (version 2.10.0 or later), you can do:
```

Comment:
Should we add a webpack example as well?

Reply:
This is for webpack as well, any of our bundler plugins can be configured like this. How could we make this clearer, without repeating this for all types of bundlers? 🤔 I added a comment above:

```js
// For example, the @sentry/webpack-plugin passed to the webpack config
```

WDYT about this?

Reply:
Ah was just thinking of the case where we don't use our bundler plugins

Reply:
I'd say for this case it's OK to link to the tree shaking page, as otherwise we need to add quite a lot of context again 😅